### PR TITLE
Catch Up performance hotfix

### PR DESCRIPTION
**Pull request by dmdashenkov**

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: server/src/main/java/org/spine3/server/event/EventStorage.java [hunk 1]
```diff
@@ -57,11 +66,31 @@ public Event apply(@Nullable EventEntity input) {
                 }
             };
 
+    @Override
+    protected org.spine3.server.storage.EventStorage createStorage(StorageFactory factory) {
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/event/EventStorage.java]: Let's avoid FQNs as much as we can.

File: server/src/main/java/org/spine3/server/event/EventStorage.java [hunk 2]
```diff
@@ -57,11 +66,31 @@ public Event apply(@Nullable EventEntity input) {
...
     Iterator<Event> iterator(EventStreamQuery query) {
-        final Iterator<EventEntity> filtered = iterator(createEntityFilter(query));
-        final List<EventEntity> entities = Lists.newArrayList(filtered);
-        Collections.sort(entities, EventEntity.comparator());
-        final Iterator<Event> result = Iterators.transform(entities.iterator(), getEventFunc());
+        final org.spine3.server.storage.EventStorage storage = recordStorage();
+        final Map<EventId, EntityRecord> records = storage.readAll(query);
+        final Collection<EventEntity> entities = transform(records.entrySet(),
+                                                           storageRecordToEntity());
+        // TODO:2017-05-19:dmytro.dashenkov: Remove after the Entity Column approach is implemented.
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/event/EventStorage.java]: @dmdashenkov please create an issue in `core-java` right after this PR is merged.

File: server/src/test/java/org/spine3/server/event/EventStoreShould.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    private static Event projectCreated(Timestamp timestamp) {
+        final ProjectCreated msg = Sample.messageOfType(ProjectCreated.class);
+        return eventFactory.createEvent(msg, null, timestamp);
+    }
+
+    @SuppressWarnings("BusyWait")
+    private static void await(AtomicBoolean done) {
+        while (!done.get()) {
+            try {
+                Thread.sleep(200);
```
Comment (armiol) [inline server/src/test/java/org/spine3/server/event/EventStoreShould.java]: Let's eliminate it, even in this PR.

File: server/src/main/java/org/spine3/server/event/EventStorage.java (hunk 1 above)
Comment (dmdashenkov) [inline server/src/main/java/org/spine3/server/event/EventStorage.java]: We should for sure but we cannot avoid usage of FQN if the qualified class has the same short name as the one that uses it.

**Comment by dmdashenkov** [comment]
@armiol, PTAL again.

**Pull request merged**